src/Symfony/Component/Serializer/composer.json

```diff
@@ -17,7 +17,8 @@
     ],
     "require": {
         "php": "^7.2.5",
-        "symfony/polyfill-ctype": "~1.8"
+        "symfony/polyfill-ctype": "~1.8",
+        "composer/semver": "^1.0@dev"
```

Comment:
this should be turned into an optional dependency IMHO, and the code should throw a LogicException when the dep is actually needed

Reply:
@nicolas-grekas I think you're right but this may be problematic, in dev composer/semver will be present because composer includes this dependency.
So in dev the exception will never be thrown but will be in production. 

How can I avoid it ?

Reply:
> in dev composer/semver will be present because composer includes this dependency.

are you sure about this?

Reply:
> are you sure about this?

https://github.com/composer/composer/blob/master/composer.json#L27